app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/BacksplashImage.scala

```diff
@@ -166,6 +170,195 @@ final case class BacksplashGeotiff(
+    corrections: ColorCorrect.Params,
+    singleBandOptions: Option[SingleBandOptions.Params],
+    projectId: UUID,
+    projectLayerId: UUID,
+    mask: Option[MultiPolygon],
+    prefix: String,
+    tracingContext: TracingContext[IO]
+)(implicit contextShift: ContextShift[IO])
+    extends MultiTiffImage[IO, IO.Par] {
+
+  val metadata = SceneMetadataFields()
+  val imageName: Option[String] = prefix.split("/").lastOption
+  val tags = Map(
+    "imageName" -> (imageName getOrElse ""),
+    "imageId" -> s"$imageId",
+    "subsetBands" -> subsetBands.mkString(","),
+    "prefix" -> prefix,
+    "readType" -> "LandsatMultitiff"
+  )
+
+  def getUri(band: Int): Option[String] = imageName map { name =>
+    s"$prefix/${name}_B${band + 1}.TIF"
+  }
+
+  def getBandRasterSource(i: Int,
+                          context: TracingContext[IO]): IO[RasterSource] = {
+    val uri = getUri(i) getOrElse { "" }
+    val rsTags = tags.combine(Map("uri" -> uri))
+    context.childSpan("getBandRasterSource", rsTags) use { _ =>
+      logger.debug(s"Using GDAL Raster Source: ${uri}")
+      // Do not bother caching - let GDAL internals worry about that
+      val rasterSource = GDALRasterSource(URLDecoder.decode(uri, "UTF-8"))
+      IO {
+        metadata.noDataValue match {
+          case Some(nd) =>
+            rasterSource.interpretAs(DoubleUserDefinedNoDataCellType(nd))
+          case _ =>
+            rasterSource
+        }
+
+      }
+    }
+  }
+
+  def selectBands(bands: List[Int]) = this.copy(subsetBands = bands)
+}
+
+sealed abstract class MultiTiffImage[F[_]: Monad, G[_]](
+    implicit F: Parallel[F, G],
+    sync: Sync[F],
+    mode: Mode[F],
+    Err: MonadError[F, Throwable]
+) extends BacksplashImage[F]
+    with LazyLogging {
+
+  implicit val tileCache = Cache.tileCache
+  implicit val rasterSourceCache = Cache.rasterSourceCache
+
+  val tags: Map[String, String]
+
+  def getRasterSource(context: TracingContext[F]): F[RasterSource] =
+    context.childSpan("getRasterSource", tags) use { child =>
+      getBandRasterSource(subsetBands.headOption getOrElse 0, child)
+    }
+
+  /** Get a single band raster source for one band of this image
+    *
+    * MultiTiff image assumes that you have a single scene split up over several
+    * single band tiffs, e.g., how Landast 8 and Sentinel-2 are stored on AWS.
+    * This assumption will cause some things to break if you end up trying to make
+    * a MultiTiffImage out of a multi-band tiff, e.g., you might think you're going
+    * to color correct one way and have something else happen entirely.
+    */
+  def getBandRasterSource(i: Int, context: TracingContext[F]): F[RasterSource]
+
+  def getBandRasterSources(
+      bs: NonEmptyList[Int],
+      context: TracingContext[F]
+  ): F[NonEmptyList[RasterSource]] =
+    bs parTraverse { band =>
+      getBandRasterSource(band, context)
+    }
+
+  def getHistogram(context: TracingContext[F]): F[Array[Histogram[Double]]] =
```

Comment:
I'm pretty unsure about it. To store them in the database would (at this point) require `BacksplashImage` to know about the database, which I think is a little unfortunate. We can move specific implementations out of `BacksplashImage` / `backsplash-core`... but then we'd lose exhaustiveness checks from the sealed trait. We could also throw them into memcached on first read instead of the db, which would trade persistence for maybe having to store histograms for a million Landsat 8 scenes.

I don't have a strong prior about which of these is likely to be best, but also none of them are That Hard :tm:, so it might make sense to try them and load test them.